repos/system_upgrade/common/actors/scansourcefiles/tests/unit_test_scansourcefiles.py

```diff
@@ -1,5 +1,163 @@
-def test_scansourcefiles():
-    # TODO(pstodulk): keeping unit tests for later after I check the idea
-    # of this actor with the team.
-    # JIRA: OAMG-10367
-    pass
+import os
+
+import pytest
+
+from leapp.libraries.actor import scansourcefiles
+from leapp.libraries.common import testutils
+from leapp.libraries.stdlib import api, CalledProcessError
+from leapp.models import FileInfo, TrackedFilesInfoSource
+
+CUR_DIR = os.path.dirname(os.path.abspath(__file__))
+FILES_DIR = os.path.join(CUR_DIR, 'files')
+
+# values in list should correspond with the return values of get_source_major_version()
+RHEL_MAJOR_VERSIONS_LIST = list(range(8, 9+1))
+
+TRACKED_FILES_BY_VERSIONS = {
+    version: [os.path.join(FILES_DIR, 'file_major_version_{}'.format(version))]
+    for version in RHEL_MAJOR_VERSIONS_LIST
+    }
+
+VERSION_FILES_MOCKED = {
+    'common': [
+        os.path.join(FILES_DIR, 'file_not_rpm_owned')
+    ]
+}
+
+VERSION_FILES_MOCKED.update(TRACKED_FILES_BY_VERSIONS)
+
+
+@pytest.mark.parametrize('run_output,expected_output', (
+    ({'exit_code': 0}, False),
+    ({'exit_code': 1, 'stdout': 'missing'}, True),
+    ({'exit_code': 1, 'stdout': '5'}, True),
+    ({'exit_code': 1, 'stdout': '5 missing'}, True),
+    ({'exit_code': 1, 'stdout': 'fi5ve'}, True),
+    ({'exit_code': 1, 'stdout': 'NoFiveNoMissing'}, False),
+))
+def test_is_modified(monkeypatch, run_output, expected_output):
+    def mocked_run(cmd, *args, **kwargs):
+        assert cmd == ['rpm', '-Vf', '--nomtime', input_file]
+        return run_output
+
+    input_file = os.path.join(FILES_DIR, 'mocked_file')
```

Comment:
Nothing really wrong with this, but you could use basically any path such as `/use/somefile` as long as it's valid. There is not need for it to be at the path you used.